### DISCUSSION TITLE
How to use "synb0" in Dipy for preprocessing

### DISCUSSION THREAD

**Discussion by barisgenc12** [dipy/dipy#2853] [3. Community Lounge]
Hello Dipy Community,

I'm currently working on preprocessing diffusion magnetic resonance imaging (dMRI) data using Dipy, and I came across the "synb0" method, which seems to be useful for distortion correction. However, I couldn't find detailed documentation or examples on how to use it properly.

Could someone please provide guidance on how to utilize the "synb0" method in Dipy for distortion correction? Specifically, I would like to know the necessary inputs, the processing steps involved, and how to interpret the results.

Any code snippets or examples would be greatly appreciated. Thank you in advance for your help!

Best regards,
Barış

**Comment by pjsjongsung** [comment]
We do not have the whole pipeline of Synb0 yet. Thus, it is necessary to have the image ready for the Synb0 model according to its pipeline.

Similar discussion is on #2811 . We'll update the whole pipeline in DIPY asap!

**Comment by smeisler** [comment]
(cross-posted from #2811)

Hi all, I would like to continue this thread.

As it stands it appears that DIPY implements the last stage of the pipeline and presumes inputs follow the the pipeline in the[ Synb0 paper](https://www.sciencedirect.com/science/article/pii/S0730725X18306179?via%3Dihub). However, after looking at the DIPY documentation, it is unclear what this means precisely. I would like to add some synb0 authors here so we can resolve some of these issues - I am having trouble tagging them here but will alert them to this thread.

From the paper:

To apply the trained networks, an MPRAGE dataset without a paired multi-shot EPI scan is affine registered (12 degree of freedom using FSL flirt [[24](https://www.sciencedirect.com/science/article/pii/S0730725X18306179?via%3Dihub#bb0120)]) to the MNI-152 1 mm isotropic T1-weighted atlas. All overlapping sets of three contiguous slices for the scan are written out (193, 229, and 193 stacks of 2D images per subject, axial, coronal, sagittal, respectively).

For anyone who needed clarification, this is MNI152NLin6Sym and not MNI152NLin2009cAsym.

For DIPY developers @pjsjongsung : In DIPY why does the image have to be 77X91X77? Unless I am misunderstanding something that appears to be in contradiction with what is described above, unless only a particular part of the neural net is implemented in DIPY. If so, what part is it?

Similarly, I would like to confirm that the following needs to take place before entering the neural net, and that this is indeed what is expected before DIPY.

Skull-stripped
Normalized
Aligned to one another in MNI152NLin6Sym space (1mm voxel for T1, what about b0)?
Next, this argument is used in Synb0-DISCO and I cannot find an analogous step in DIPY:

acqparams.txt: A text file that describes the acqusition parameters, and is described in detail on the FslWiki for topup (https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/topup). Briefly, it describes the direction of distortion and tells TOPUP that the synthesized image has an effective echo spacing of 0 (infinite bandwidth). An example acqparams.txt is displayed below, in which distortion is in the second dimension, note that the second row corresponds to the synthesized, undistorted, b0:

$ cat acqparams.txt 
0 1 0 0.062
0 1 0 0.000

Is this information not used in DIPY, or is it presumed from something?

Thanks for your help,
Steven
